### PR TITLE
Cancelling the resource in case of tarball error

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/7359

### CODE REVIEW

File: service/controller/chart/v1/resource/release/desired.go [hunk 1]
```diff
@@ -27,7 +28,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
 	if err != nil {
-		return nil, microerror.Mask(err)
+		r.logger.LogCtx(ctx, "level", "error", "message", "pulling chart failed", "stack", microerror.Stack(err))
```
Comment (tomahawk28) [inline service/controller/chart/v1/resource/release/desired.go]: Putting a stack in the end since network error could vary.

Comment (xh3b4sd) [reply service/controller/chart/v1/resource/release/desired.go]: When we expect an error the logging should only be `warning`.

File: service/controller/chart/v1/resource/release/desired.go [hunk 2]
```diff
@@ -27,7 +28,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (tomahawk28) [reply service/controller/chart/v1/resource/release/desired.go]: In this case, it's `executionFailedError` in `helmclient`. This is one of the network error and the fact that we can't use the matcher against `executionFailedError`, Maybe I should return `pullChartFailedError` as a customized one in `helmclient`, WDYT?
https://github.com/giantswarm/helmclient/blob/46206f66eaf4a9a4aa526457ae5d8e0124ec3d90/pull_chart_tarball.go#L39-L41

[the remark this answers is not in the document]
Comment (xh3b4sd) [reply service/controller/chart/v1/resource/release/desired.go]: 100%, let's get more fine grained error handling here down the line.

[the remark this answers is not in the document]
Comment (rossf7) [reply service/controller/chart/v1/resource/release/desired.go]: Yes I think returning pullChartFailedError is good.

**Comment by tomahawk28** [comment]
@giantswarm/team-batman @giantswarm/sig-operator It's ready, PTAL

File: service/controller/chart/v1/resource/release/create.go [hunk 3]
```diff
@@ -29,7 +29,12 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 
 		tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
 		if err != nil {
```
Comment (xh3b4sd) [inline service/controller/chart/v1/resource/release/create.go]: Why is this still matching all errors? We said this does not make sense.

Comment (tomahawk28) [reply service/controller/chart/v1/resource/release/create.go]: I didn't see there are other two usage 🤦‍♂ Sorry.

**Pull request merged**